### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by LordSimal**
Refs: #11

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -30,16 +30,18 @@
...
+		"psalm": "tools/psalm --show-info=false",
 		"stan": [
 			"@phpstan",
 			"@psalm"
 		],
+		"stan-baseline": "tools/phpstan --generate-baseline",
+		"psalm-baseline": "tools/psalm --set-baseline=psalm-baseline.xml",
 		"test": "phpunit",
-		"stan-setup": "cp composer.json composer.backup && composer require --dev phpstan/phpstan:~1.9.0 psalm/phar:~5.4.0 && mv composer.backup composer.json",
+		"stan-setup": "phive install",
```
Comment (LordSimal) [inline composer.json:44]: I've introduced the phive toolset which we also use in the core.
See https://github.com/cakephp/cakephp/blob/5.x/.github/CONTRIBUTING.md#test-cases-codesniffer-and-static-analysis to learn more about it

File: tests/bootstrap.php [hunk 2]
```diff
@@ -32,86 +25,4 @@
...
-        'url' => getenv('DB_URL_COMPARE'),
-    ]);
-}
-
-// Create test database schema
-if (env('FIXTURE_SCHEMA_METADATA')) {
-    $loader = new SchemaLoader();
-    $loader->loadInternalFile(env('FIXTURE_SCHEMA_METADATA'), 'test');
-}
+require $root . '/vendor/cakephp/cakephp/tests/bootstrap.php';
```
Comment (LordSimal) [inline tests/bootstrap.php:28]: If this plugin doesn't need a custom test bootstrapping process, then just requiring the core one is fine

File: src/Model/Entity/Asset.php [hunk 3]
```diff
@@ -313,7 +315,7 @@ public function copyToWebroot(?string $filename = null, string $path = 'files',
 
         $prefix = null;
         if (empty($config['no_prefix'])) {
-            $prefix = Strings::substring(md5($this->modified->toDateTimeString()), 0, 3) . '-';
+            $prefix = Strings::substring(md5($this->modified?->toDateTimeString() ?? ''), 0, 3) . '-';
```
Comment (LordSimal) [inline src/Model/Entity/Asset.php:318]: this should be looked into.
As said in another issue `modified` shouldn't be nullable in the first place.

File: src/Controller/Admin/AssetsController.php [hunk 4]
```diff
@@ -122,7 +127,7 @@ public function download(string $id): Response
         });
 
         $response = $this->getResponse()
-            ->withType($asset->mimetype)
+            ->withType($asset->mimetype ?? 'jpg')
```
Comment (LordSimal) [inline src/Controller/Admin/AssetsController.php:130]: What if the asset doesn't have a mime type? Currently `$asset->mimetype` is nullable.

File: tests/Utilities/ImageAssetTest.php [hunk 5]
```diff
@@ -11,18 +11,10 @@ class ImageAssetTest extends TestCase
 {
     protected const DATA_IMAGES = __DIR__ . DS;
 
-    protected function setUp(): void
-    {
-        parent::setUp();
-        if (!defined('PLUGIN_ROOT')) {
-            include __DIR__ . DS . '..' . DS . 'bootstrap.php';
```
Comment (passchn) [inline tests/Utilities/ImageAssetTest.php:18]: I can't run tests from the phpstorm sidebar without this as the bootstrap script seems not to be run then.

**Comment by passchn** [comment]
Thank you so much! Tests work for me locally when running with composer, but not through the phpstorm sidebar.. Should be due to the bootstrap.php not being loaded automatically although it being defined in the phpunit config file. 

Also thanl you for fixing types and code style. Regarding the nullable fields: This would require a migration so I wanted to wait until I might release some further db changes..

File: tests/Utilities/ImageAssetTest.php (hunk 5 above)
Comment (LordSimal) [reply tests/Utilities/ImageAssetTest.php:18]: This is a bug in PHPStrom.
You can manually tell PHPStorm to use `phpunit.xml.dist` like so

image

Comment (passchn) [reply tests/Utilities/ImageAssetTest.php:18]: Ah... now it works :) 

Mittel (Bildschirmfoto 2023-09-23 um 21 15 33)

**Comment by passchn** [comment]
Was released in version 0.5.1

**Pull request merged**